dash/include/dash/atomic/GlobAtomicRef.h

```diff
@@ -181,7 +181,10 @@ class GlobRef<dash::Atomic<T>>
                        reinterpret_cast<void * const>(&result),
                        dash::dart_punned_datatype<T>::value,
                        DART_OP_NO_OP);
-    dart_flush_local(_gptr);
+    // a flush_local is not sufficient as some versions of
+    // openmpi and MPICH / IntelMPI fetch the old value
+    // due to progress problems
```

Comment:
This is a bug and if I remember correctly you already reported it to OpenMPI. In fact this is your code and I took it from the reproducer. Unfortunately I did not document the issue.

Reply:
Ahh I remember. I reported this and it should be fixed since Open MPI 2.1.1 I think. But that was not related correctness, `MPI_Win_flush_local` didn't trigger the progress engine back then.

Reply:
Ok, I reverted this back to the version in development and it seems to work now. Regarding correctness: This only happend if spinning on a local (atomic) value.